commit 2eaf8bd1b73a48bf16baa386e409cfcd8614787a
Author: Alex Wilson <alex@cooperi.net>
Date:   Mon Jan 24 10:42:54 2022 +1000

    xxx: illumos patches

diff --git a/src/usr.sbin/bgpctl/bgpctl.c b/src/usr.sbin/bgpctl/bgpctl.c
index 724c185f..8d7c0bd3 100644
--- src/usr.sbin/bgpctl/bgpctl.c
+++ src/usr.sbin/bgpctl/bgpctl.c
@@ -75,6 +75,8 @@ usage(void)
 	exit(1);
 }
 
+#undef	sun
+
 int
 main(int argc, char *argv[])
 {
diff --git a/src/usr.sbin/bgpd/control.c b/src/usr.sbin/bgpd/control.c
index 844a05e5..7d11d030 100644
--- src/usr.sbin/bgpd/control.c
+++ src/usr.sbin/bgpd/control.c
@@ -39,6 +39,8 @@ int		 control_close(struct ctl_conn *);
 void		 control_result(struct ctl_conn *, u_int);
 ssize_t		 imsg_read_nofd(struct imsgbuf *);
 
+#undef sun
+
 int
 control_check(char *path)
 {
diff --git a/src/usr.sbin/bgpd/session.c b/src/usr.sbin/bgpd/session.c
index 61224667..76c50390 100644
--- src/usr.sbin/bgpd/session.c
+++ src/usr.sbin/bgpd/session.c
@@ -1141,6 +1141,7 @@ session_setup_socket(struct peer *p)
 			 */
 			if (p->conf.ttlsec) {
 				ttl = 256 - p->conf.distance;
+#if defined(IP_MINTTL)
 				if (setsockopt(p->fd, IPPROTO_IP, IP_MINTTL,
 				    &ttl, sizeof(ttl)) == -1) {
 					log_peer_warn(&p->conf,
@@ -1148,6 +1149,7 @@ session_setup_socket(struct peer *p)
 					    "setsockopt MINTTL");
 					return (-1);
 				}
+#endif
 				ttl = 255;
 			}
 
